includes/class-stats.php

```diff
@@ -309,38 +401,66 @@ private function get_registered_stats() {
 				'job_listing_view'                 => [
 					'log_callback' => [ $this, 'log_stat' ], // Example of overriding how we log this.
 					'trigger'      => 'page-load',
+					'page'         => 'listing',
 				],
 				'job_listing_view_unique'          => [
 					'unique'          => true,
 					'unique_callback' => [ $this, 'unique_by_post_id' ],
 					'trigger'         => 'page-load',
+					'page'            => 'listing',
 				],
 				'job_listing_apply_button_clicked' => [
 					'trigger'         => 'apply-button-clicked',
 					'element'         => 'input.application_button',
 					'event'           => 'click',
 					'unique'          => true,
 					'unique_callback' => [ $this, 'unique_by_post_id' ],
+					'page'            => 'listing',
+				],
+				'jobs_view'                        => [
+					'trigger' => 'page-load',
+					'page'    => 'jobs',
+				],
+				'jobs_view_unique'                 => [
+					'trigger'         => 'page-load',
+					'page'            => 'jobs',
+					'unique'          => true,
+					'unique_callback' => [ $this, 'unique_by_post_id' ],
+				],
+				// New style of declaration, a stat that relies on calling a custom js func.
+				'job_listing_impressions'          => [
+					'trigger'         => 'job-listing-impression',
+					'unique'          => true,
+					'js_callback'     => 'WPJMStats.initListingImpression',
+					'unique_callback' => [ $this, 'unique_by_post_id' ],
+					'page'            => 'jobs',
 				],
 			]
 		);
 	}
 
 	/**
-	 * Prepare stats for wp_localize.
+	 * Determine what stats should be added to the kind of page the user is viewing.
 	 *
-	 * @param int $post_id Optional post id.
+	 * @param int    $post_id Optional post id.
+	 * @param string $page    The page in question.
 	 *
 	 * @return array
 	 */
-	private function get_stats_for_ajax( $post_id = 0 ) {
+	private function get_stats_for_ajax( $post_id = 0, $page = 'listing' ) {
 		$ajax_stats = [];
 		foreach ( $this->get_registered_stats() as $stat_name => $stat_data ) {
+			if ( $page !== $stat_data['page'] ) {
```

Comment:
If we don't split up the stats data, let's use `array_filter` here, so we don't have to copy the fields below.